src/main/cpp/Intake.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2023-2024 CurtinFRC
+// Open Source Software, you can modify it according to the terms
+// of the MIT License at the root of this project
+
+#include "Intake.h"
+
+Intake::Intake(IntakeConfig config) : _config(config), _pid(config.path + "/pid", config.pidConfig) {}
+
+IntakeConfig Intake::GetConfig() {
+  return _config;
+}
+
+void Intake::OnStart() {
+  _pid.Reset();
+}
+
+void Intake::OnUpdate(units::second_t dt) {
+
+  switch (_state) {
+    case IntakeState::kIdle: 
+    {
+      if (_config.intakeSensor->Get() == false) { 
+        SetState(IntakeState::kHold);
+      }
+      _stringStateName = "Idle";
+      _pid.Reset();
+      _setVoltage = 0_V;
+    } 
+    break;
+
+    case IntakeState::kRaw: 
+    {
+      _stringStateName = "Raw";
+      _pid.Reset();
+      _setVoltage = _rawVoltage;
+    } 
+    break;
+
+    case IntakeState::kEject: 
+    {
+      _stringStateName = "Eject";
+      _setVoltage = 7_V;
+      _pid.Reset();
+      if (_config.intakeSensor->Get() == true) {
+        SetState(IntakeState::kIdle); 
+      }
+    } 
+    break;
+
+    case IntakeState::kHold: 
+    {
+      _stringStateName = "Hold";
+
+      _setVoltage = 0_V;
+
+      if (_config.intakeSensor->Get() == true) {
+        SetState(IntakeState::kIdle);
+      }
+      
+      _pid.SetSetpoint(_goal);
```

Comment:
because goal is never going to change we might just want to set the setpoint to 0 instead of the variable, so it can't accidently be changed in the behaviour